test/e2e/kubevirt.go

```diff
@@ -1420,4 +1429,145 @@ passwd:
+					cidr:      correctCIDRFamily(cidrIPv4, cidrIPv6),
+					role:      "primary",
+				})
+			By("Creating NetworkAttachmentDefinition")
+			nad := generateNAD(netConfig)
+			Expect(crClient.Create(context.Background(), nad)).To(Succeed())
+
+			kubevirtPod := &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "virt-launcher-vm1",
+					Namespace: namespace,
+					Labels: map[string]string{
+						kubevirtv1.VirtualMachineNameLabel: "vm1",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Name:  "compute",
+						Image: "quay.io/nmstate/c10s-nmstate-dev:latest",
+						SecurityContext: &corev1.SecurityContext{
+							Privileged: ptr.To(true),
+							Capabilities: &corev1.Capabilities{
+								Add: []corev1.Capability{"NET_ADMIN"},
+							},
+						},
+					}},
+				},
+			}
+			By("Create virt-launcher pod")
+			Expect(crClient.Create(context.Background(), kubevirtPod)).To(Succeed())
+
+			By("Wait for virt-launcher pod to be ready and primary UDN network status to pop up")
+			waitForPodsCondition([]*corev1.Pod{kubevirtPod}, func(g gomega.Gomega, pod *corev1.Pod) {
+				ok, err := testutils.PodRunningReady(pod)
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(ok).To(BeTrue())
+
+				primaryUDNNetworkStatuses, err := podNetworkStatus(pod, func(networkStatus nadapi.NetworkStatus) bool {
+					return networkStatus.Name == "ovn-kubernetes" && networkStatus.Default && networkStatus.Interface == "ovn-udn1"
+				})
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(primaryUDNNetworkStatuses).To(HaveLen(1))
+				primaryUDNNetworkStatus = primaryUDNNetworkStatuses[0]
+			})
+
+			By("Wait NetworkManager readiness")
+			Eventually(func() error {
+				_, err := virtLauncherCommand("systemctl is-active NetworkManager")
+				return err
+			}).
+				WithTimeout(5 * time.Second).
+				WithPolling(time.Second).
+				Should(Succeed())
+
+			By("Reconfigure primary UDN interface to use dhcp/nd for ipv4 and ipv6")
+			_, err := virtLauncherCommand(`
+nmcli c mod ovn-udn1 ipv4.addresses "" ipv6.addresses "" ipv4.gateway "" ipv6.gateway "" ipv6.method auto ipv4.method auto ipv6.addr-gen-mode eui64
+nmcli d reapply ovn-udn1`)
+			Expect(err).ToNot(HaveOccurred())
+
+		})
+		It("should configuration IPv4 and IPv6 using DHCP and NDP", func() {
+
+			dnsService, err := fr.ClientSet.CoreV1().Services(config.Kubernetes.DNSServiceNamespace).
+				Get(context.Background(), config.Kubernetes.DNSServiceName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			if isIPv4Supported() {
+				expectedIP, err := util.MatchIPStringFamily(false /*ipv4*/, primaryUDNNetworkStatus.IPs)
+				Expect(err).ToNot(HaveOccurred())
+
+				expectedDNS, err := util.MatchIPStringFamily(false /*ipv4*/, dnsService.Spec.ClusterIPs)
+				Expect(err).ToNot(HaveOccurred())
+
+				cidrIP, _, err := net.ParseCIDR(cidrIPv4)
+				Expect(err).ToNot(HaveOccurred())
+				expectedGatway := iputils.NextIP(cidrIP).String()
```

Comment:
please export to an helper. Or just use https://github.com/ovn-org/ovn-kubernetes/blob/ef04a67650d3a7048de3525512ba514cdfb31ff0/go-controller/pkg/util/net.go#L38